js/src/logger.ts

```diff
@@ -2863,6 +3000,39 @@ function validateAndSanitizeExperimentLogPartialArgs(
   }
 }
 
+/**
+ * Helper function for uploading attachments. Recursively extracts `Attachment` values.
+ * @param event The event to be extract. Will be modified.
+ * @returns The function to get the modified event and the extracted `Attachment` objects.
+ */
+function extractAttachments<T extends Partial<BackgroundLogEvent>>(
```

Comment:
I wonder if the extra type safety is actually important here. Since we're mutating the object in place, could we just make it `event: Record<string, unknown>`?

Reply:
I am closing this comment because we no longer mutate the object in place.